Assets/Features/Building/Scripts/Grid/Grid.cs

```diff
@@ -309,6 +337,11 @@ public bool Remove(Vector3Int gridVector)
                 //Remove from array
                 foreach (var item in group)
                 {
+                    CellData cell = _cells[item.x, item.y, item.z];
+
+                    if (_atlas.Items[cell.Tile].UtilityType != UtilityType.None)
+                        _utilityLocations[_atlas.Items[cell.Tile].UtilityType].Remove(item);
```

Comment:
This is incredibly hard to read. Not necessary for now, but for readability sake, try to improve readability.